docs/modules/ROOT/pages/guide-metatx.adoc

```diff
@@ -0,0 +1,349 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/metatx/ERC2771Context.sol";
+import "@openzeppelin/contracts/metatx/MinimalForwarder.sol";
+
+contract Registry is ERC2771Context {  
+  event Registered(address indexed who, string name);
+
+  mapping(address => string) public names;
+  mapping(string => address) public owners;
+
+  constructor(MinimalForwarder forwarder) // Initialize trusted forwarder
+    ERC2771Context(address(forwarder)) {
+  }
+
+  function register(string memory name) external {
+    require(owners[name] == address(0), "Name taken");
+    address owner = _msgSender(); // Changed from msg.sender
+    owners[name] = owner;
+    names[owner] = name;
+    emit Registered(owner, name);
+  }
+}
+```
+
+Run `npx hardhat compile` to get the code ready for deployment.
+
+[[deploy-using-relayer]]
+== Deploy Using Relayer
+
+You can easily deploy a compiled smart contract without handling a private key by using https://www.npmjs.com/package/defender-relay-client[Defender Relayer client].
+
+Use the following script to deploy the contract using the Relayer just created:
+
+```
+require('dotenv').config();
+const { DefenderRelayProvider, DefenderRelaySigner } = require('defender-relay-client/lib/ethers');
+const { ethers } = require('hardhat');
+const fs = require('fs');
+
+async function main() {
+  const { secretKey: {apiKey, secretKey} } = JSON.parse(fs.readFileSync('.env.relay'))
+  const credentials = { apiKey, apiSecret: secretKey };
+  const provider = new DefenderRelayProvider(credentials);
+  const relaySigner = new DefenderRelaySigner(credentials, provider, { speed: 'fast' });
+ 
+  const Forwarder = await ethers.getContractFactory('MinimalForwarder');
+  const forwarder = await Forwarder.connect(relaySigner).deploy().then(f => f.deployed());
+
+  const Registry = await ethers.getContractFactory("Registry");
+  const registry = await Registry.connect(relaySigner).deploy(forwarder.address).then(f => f.deployed());
+
+  fs.writeFileSync('deploy.json', JSON.stringify({
+    MinimalForwarder: forwarder.address,
+    Registry: registry.address,
+  }, null, 2));
+
+  console.log(`MinimalForwarder: ${forwarder.address}\nRegistry: ${registry.address}`);
+}
+
+if (require.main === module) {
+  main().then(() => process.exit(0))
+    .catch(error => { console.error(error); process.exit(1); });
+}
+
+```
+
+This code pulls the Relayer's credentials from the local file along with the artifacts for the `Registry` and `MinimalForwarder` contracts and uses ethers.js to deploy. The relevant addresses of these contracts are saved to the local file `deploy.json`.
+
+In the demo app, run this script with `yarn deploy`
+
+Note: As the owners of the Relayer, are the paying for the gas, so it is necessary to keep the Relayer funded sufficiently to cover future transactions. For Defender users on testnets, Relayer balances are auto-topped as necessary to ensure transactions go through.
+
+
+[[create-autotask]]
+== Create Autotask via API
```

Comment:
We very frequently see users that try to expose their relayer directly to the frontend, bypassing the autotask. We should explain here why it's important to have the autotask: so you can decide whether to pay for the request or not. And that if you expose the relayer directly, anyone can just get your relayer keys and use the relayer to send whatever tx they want.